code/modules/vending/_vending.dm

```diff
@@ -320,47 +294,61 @@ IF YOU MODIFY THE PRODUCTS LIST OF A MACHINE, MAKE SURE TO UPDATE ITS RESUPPLY C
 	return ..()
 
 /obj/machinery/vending/ui_interact(mob/user)
+	var/onstation = FALSE
+	for(var/A in SSmapping.levels_by_trait(ZTRAIT_STATION))
+		if(src.z == A)
+			onstation = TRUE
 	var/dat = ""
-
-	dat += "<h3>Select an item</h3>"
-	dat += "<div class='statusDisplay'>"
-	if(!product_records.len)
-		dat += "<font color = 'red'>No product loaded!</font>"
-	else
-		var/list/display_records = product_records
-		if(extended_inventory)
-			display_records = product_records + hidden_records
-		if(coin || bill)
-			display_records = product_records + coin_records
-		if((coin || bill) && extended_inventory)
-			display_records = product_records + hidden_records + coin_records
-		dat += "<ul>"
-		for (var/datum/data/vending_product/R in display_records)
-			dat += "<li>"
-			if(R.amount > 0)
-				dat += "<a href='byond://?src=[REF(src)];vend=[REF(R)]'>Vend</a> "
+	var/datum/bank_account/account
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		var/obj/item/card/id/C = H.get_idcard()
+		if(!C)
+			dat += "<font color = 'red'><h3>No ID Card detected!</h3></font>"
+		else if (!C.registered_account)
+			dat += "<font color = 'red'><h3>No account on registered ID card!</h3></font>"
+		if((onstation && C && C.registered_account) || (!onstation))
+			if(onstation)
+				account = C.registered_account
+			dat += "<h3>Select an item</h3>"
+			dat += "<div class='statusDisplay'>"
+			if(!product_records.len)
+				dat += "<font color = 'red'>No product loaded!</font>"
 			else
-				dat += "<span class='linkOff'>Sold out</span> "
-			dat += "<font color = '[R.display_color]'><b>[sanitize(R.name)]</b>:</font>"
-			dat += " <b>[R.amount]</b>"
-			dat += "</li>"
-		dat += "</ul>"
-	dat += "</div>"
-	if(premium.len > 0)
-		dat += "<b>Change Return:</b> "
-		if (coin || bill)
-			dat += "[(coin ? coin : "")][(bill ? bill : "")]&nbsp;&nbsp;<a href='byond://?src=[REF(src)];remove_coin=1'>Remove</a>"
-		else
-			dat += "<i>No money</i>&nbsp;&nbsp;<span class='linkOff'>Remove</span>"
-	if(istype(src, /obj/machinery/vending/snack))
-		dat += "<h3>Chef's Food Selection</h3>"
-		dat += "<div class='statusDisplay'>"
-		for (var/O in dish_quants)
-			if(dish_quants[O] > 0)
-				var/N = dish_quants[O]
-				dat += "<a href='byond://?src=[REF(src)];dispense=[sanitize(O)]'>Dispense</A> "
-				dat += "<B>[capitalize(O)]: [N]</B><br>"
-		dat += "</div>"
+				var/list/display_records = product_records + coin_records
+				if(extended_inventory)
+					display_records = product_records + coin_records + hidden_records
+				dat += "<ul>"
+				for (var/datum/data/vending_product/R in display_records)
+					var/price_listed = "$[default_price]"
```

Comment:
Add a proc for later use for people to implement individual item prices. Something like `item.get_price()`. People can implement prices later and argue over individual ones but get the framework in there and have the machine properly use it even if everything just returns a default value for now.

Reply:
Or in the case of newed things you can have a default price var on the item

Reply:
added